linode/manager
# tech story: [M3-8385] - Replace lodash set utility function to handle prototype pollution security threat

## Description 📝
The lodash.set package causes a [security concern for prototype pollution](https://github.com/linode/manager/security/dependabot/99). This PR aims to resolve the security concern by creating our own version of set (specific to our use cases, not a rehash of lodash's set) that is also prototype pollution safe -- see internal ticket for full details on reasoning behind this approach, but the tldr:
- Although the security issue is fixed in lodash v4.17.19+, we've already [moved away](https://github.com/linode/manager/pull/10386) from using the entire lodash package so it didn't really make sense to add it back in 
- we're already using the most updated version of lodash.set (4.3.2) that has this concern 
- so ~~I tried to recreate lodash's set, but it isn't a complete recreation of it - there's a lot of edge cases (see skipped tests)~~. we've made a simpler version of set that is specific to our use case (for example, while lodash's set might accept paths in array format, we only accept paths in string format)

## Changes  🔄
- remove lodash.set dependency 
- create our version of set
- add a bunch of tests
- removed case of importing omit from the lodash package + replaced with our already existing omit function

## Target release date 🗓️
?

## How to test 🧪

### Verification steps
- confirm `yarn test formikErrorUtils.test.ts` and that there are no UI changes for places that use `getFormikErrorsFromAPIErrors` (specifically PlacementGroups stuff!)
- ~~confirm that `yarn test utilities/set.test.ts` passes if you switch out the new version of set for lodash.set~~ (removing because this is no longer a direct copy for lodash's set)
- spotcheck the Linode Create v2 flow, since I exchanged a function there (but the one I exchanged it for is a direct replacement so it should be fine ✌️) 

## As an Author I have considered 🤔

*Check all that apply*
